### PR TITLE
Added warning message when there isn't a payment method set up

### LINKED ISSUE #4596: Task List - Payments task: add alert message
**Description:** we want to reinforce the importance of setting up a payment method and increase user intent by adding an alert message about this topic

**Acceptance Criteria:**
- [ ] Add an alert message on each page of the Payments task reinforcing the fact that enabling a payment method is required to start accepting orders 

**Event tracking:**
NA 

**Additional links:**
[Design mockup](https://www.figma.com/file/JH9XMFUCOjfXdr3N09AHRD/On-boarding-iterations-June-'20?node-id=111%3A483)

### PR DESCRIPTION
**Pull request by octaedro**
Fixes #4596

This PR adds a warning message for each page of the Payments task if payment methods are incomplete

### Accessibility

- [x] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [x] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [x] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### Screenshots
screenshot-one wordpress test-2020 07 08-16_22_11

### Detailed test instructions:

- Go to the page: `Payment methods` (URL `/wp-admin/admin.php?page=wc-settings&tab=checkout`).
- Be sure, there isn't any payment method selected.
- Go to `home screen` / `dashboard` and press `Set up payments`.
- Verify the warning `Customers cannot place orders at your store until you setup and enable a payment method.` is visible.

Screen Capture on 2020-07-08 at 18-04-23

- Go back to `home screen` / `dashboard` and verify the message is not visible anymore.

- Go to the `Payment methods` again and select a payment method.

- Go to `home screen` / `dashboard` and press `Set up payments`.
- Verify the warning message is not visible.

- Verify the message continues to look as expected in different screen widths.

### Changelog Note:
Dev: Added warning message when there isn't a payment method set up

### CODE REVIEW

File: client/layout/store-notices/index.js
Comment (timmyc) [inline client/layout/store-notices/index.js:40]: This string needs to be translated.

File: client/layout/store-notices/index.js
Comment (timmyc) [inline client/layout/store-notices/index.js:18]: Since the options store is already on `@wordpress/data` seems like we should be able to use `import { withSelect } from '@wordpress/data';` instead as an external dependency?

**Review by timmyc — CHANGES_REQUESTED** [review]
This is testing out nicely for me - responds immediately to the stat change as I interact with toggles on the payments task. Also looking good on narrow viewports - nice!

Couple of quick items need to be addressed, but should be good to go after that.

File: client/layout/store-notices/index.js
Comment (octaedro) [reply client/layout/store-notices/index.js]: Hi @timmyc, thank you for your review.
Nice catch! I added it in the commit d9c0a3d

File: client/layout/store-notices/index.js
Comment (octaedro) [reply client/layout/store-notices/index.js]: Good point! I modified it in the commit 96194bc

File: client/layout/store-notices/index.js
Comment (becdetat) [inline client/layout/store-notices/index.js:31]: should be "showMissingPaymentMethod"

File: client/layout/store-notices/index.js
Comment (becdetat) [inline client/layout/store-notices/index.js:63]: I found that I had four options returned here, even after removing all of the payment provider plugins I had sitting around. I'm guessing that the settings option doesn't get removed when removing the plugin. Deleting the options made this work. You might want to see if there's a different way to check if a payment provider is installed - maybe using `getActivePlugins()` - https://github.com/woocommerce/woocommerce-admin/blob/main/client/homescreen/stats-overview/install-jetpack-cta.js#L165

File: client/layout/store-notices/index.js
Comment (becdetat) [inline client/layout/store-notices/index.js:23]: Could you do this in the `withSelect` below, so that you do all the logic in one place and just get a `showMissingPaymentMethod` prop? That would make it much easier to refactor when another warning gets added.

Comment (octaedro) [reply client/layout/store-notices/index.js]: Good point! I moved that logic in the commit 44ed4c7

File: client/layout/store-notices/index.js
Comment (octaedro) [reply client/layout/store-notices/index.js]: Nice catch @becdetat, I corrected it in the commit 31d418f

File: client/layout/store-notices/index.js
Comment (octaedro) [reply client/layout/store-notices/index.js:56]: Hi @becdetat, thank you for your feedback.
Your idea sounds good to me.

Just to confirm:
if the user has selected one of the following payment methods:  `PayPal, PayFast, Karna Checkout, Stripe, and Square` we should verify if the plugins related to those methods are also activated. If they aren't activated we'll assume the user doesn't want those methods, even though they appear as active in our options. Did I understand correctly?

Thinking about this, I wonder if we should also update the state of the inactive payment gateways to not have inconsistencies with the information we have in the DB.

Comment (timmyc) [reply client/layout/store-notices/index.js:56]: Really good point @becdetat - that hardcoded list of options caused me to pause a bit when I saw it too.

What do you all think about possibly just calling out to the core WC payment gateways endpoint: https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-payment-gateways

That might require a bit more work though since I don't think we have a data store setup for that yet, but maybe an [apiFetch](https://developer.wordpress.org/block-editor/packages/packages-api-fetch/) could work. 

Alternatively, we could consider this kind of an edge case for now - especially given the context of this warning being shown in the Setup Checklist.

**Comment by timmyc** [comment]
Hey @jameskoster - I was hoping you could have a quick look here. So this is functioning per the spec, there are some interesting edge case scenarios that @becdetat pointed out where we could possibly make this a bit more resilient on the code side of things... but what is causing me to pause on shipping this in 1.4 is just the introduction of the contrasting alerts ( as you can see in the screenshot above ).

This very much echoes some of the feedback we got during the usability research for the new nav - but adding in another notification style with the core notif here, alongside the existing `<StoreAlert />` just to me looks a little _off_ visually.

While I appreciate the messaging we are trying to deliver in this PR, I'm wondering if we should invest some time in refactoring the `<StoreAlert />`s prior to begin shipping this form of messaging in wc-admin.

**Comment by timmyc** [comment]
After a [conversation with Jay in slack](https://a8c.slack.com/archives/C9K5T0XSP/p1594919807125300) we are going to hold off on merging this one for 1.4 - and maybe seek to align the existing StoreAlerts with the core alert styling to make the experience a bit more visually cohesive. I think we should keep this branch around, but just give it a Status Blocked for now.

File: client/layout/store-notices/index.js
Comment (samueljseay) [reply client/layout/store-notices/index.js]: @octaedro also `isVisibleMissingPaimentWarning` should be `isVisibleMissingPaymentWarning`

**Comment by pmcpinto** [comment]
@timmyc can we already implement this warning message?

**Comment by timmyc** [comment]
While I like the spirit of this feature, I'm wondering if we should reconsider the presentation/design side of this with the new homescreen "task" area in mind. To me it seems like surfacing this message as an important "task" on the homescreen might be more valuable as it would cover use cases where someone has possibly dismissed the setup checklist, or is just simply ignoring it completely.

My primary concern with this design is it just creates yet another alert/message design in an already crowded notif world ( admin alerts, store alerts, inbox messages etc ).

**Comment by pmcpinto** [comment]
Good point. @elizaan36 what do you think?

**Comment by elizaan36** [comment]
Is this a problem we've discovered from support tickets or issues from users that dismiss the task list and then don't know how to set up payment methods? I'm trying to understand where the need came from and the extent of the problem to understand the best route to take.

**Comment by pmcpinto** [comment]
> Is this a problem we've discovered from support tickets or issues from users that dismiss the task list and then don't know how to set up payment methods? I'm trying to understand where the need came from and the extent of the problem to understand the best route to take.

AFAIR it was to make sure that users really understand that they need to enable a payment option to get paid. FOMO effect

**Comment by elizaan36** [comment]
ah ok! In that case I'd agree with Timmy 

> While I like the spirit of this feature, I'm wondering if we should reconsider the presentation/design side of this with the new homescreen "task" area in mind.

If we haven't experienced issues or tickets arising from users having trouble with this, it definitely feels like something we can solve for with highlighting the payments setup task in some way rather than presenting a notice.

**Pull request closed without merging**
